src/mlpack/methods/decision_tree/decision_tree_impl.hpp

```diff
@@ -1043,12 +1091,40 @@ void DecisionTree<FitnessFunction,
   }
 
   // Now normalize into probabilities.
-  classProbabilities /= UseWeights ? sumWeights : labels.n_elem;
+  // classProbabilities /= UseWeights ? sumWeights : labels.n_elem;
```

Comment:
I'd really like if we could avoid doing this.  This means that when we do prediction, we'll have to do a division.  And I feel pretty strongly that we need to keep prediction as fast as possible.

Here is a different approach---when building the tree, allow the `PruningMethod` to store some state about each node.  Maybe a `std::map<DecisionTree*, SomeValuesStruct>` or something like this, where you can store the class counts for later pruning.  Then when training is done, the extra data held by the `PruningMethod` can be deleted, and pruning methods can collect any data they need during training.  I would say, then, that we could add a `CreateNode(...)` function to each `PruningMethod`, which could then update any internal state and variables that need to be known for training.

I know this involves a lot of refactoring, and I am sorry about that, but again, speed is really important here.